### PR TITLE
Satoshi denominated tpos

### PR DESCRIPTION
**Pull request by iWarpBTC**
I wanted it (#58) so I tried to do it. Some really straighforward changes to quickly accomplish denomination in satoshi.

### CODE REVIEW

File: templates/tpos/tpos.html [hunk 1]
```diff
@@ -755,28 +759,47 @@ <h5 class="q-mt-none q-mb-sm">
...
+          )
+        }
       },
       amountWithTipFormatted: function () {
-        return LNbits.utils.formatCurrency(
-          (this.amount + this.tipAmount).toFixed(2),
-          this.currency
-        )
+        if (this.currency == 'satoshi') {
+          return LNbits.utils.formatSat(this.amount.toFixed(0)) + ' sat'
```
Comment (prusnak) [inline templates/tpos/tpos.html:796]:
```suggestion
          return LNbits.utils.formatSat(this.amount) + ' sat'
```

File: templates/tpos/tpos.html [hunk 2]
```diff
@@ -996,7 +1021,12 @@ <h5 class="q-mt-none q-mb-sm">
       },
       submitForm: function () {
         if (this.total != 0.0) {
-          this.stack = Array.from(String(Math.ceil(this.total * 100)), Number)
+          this.stack = Array.from(
+            String(
+              Math.ceil(this.total * (this.currency == 'satoshi' ? 1 : 100))
+            ),
+            Number
+          )
```
Comment (prusnak) [inline templates/tpos/tpos.html:1029]: I think this is easier to read:

```suggestion
          if (this.currency == 'satoshi') {
              this.stack = Array.from(String(Math.ceil(this.total), Number)
          } else {
              this.stack = Array.from(String(Math.ceil(this.total * 100)), Number)
          }
```

**Comment by iWarpBTC** [comment]
Thanks for the review and suggestions!

**Review by talvasconcelos — APPROVED** [review]
Seems ok for me.

**Comment by motorina0** [comment]
The format logic is replicated in 6 places. It should be extracted to one util method:

```js
function formatAmount(amount, currency) {
  if (currency == "satoshi") {
    return LNbits.utils.formatSat(amount) + " sat";
  }
  return LNbits.utils.formatCurrency(Number(amount).toFixed(2), currency);
}
```

File: templates/tpos/index.html [hunk 3]
```diff
@@ -591,6 +591,7 @@ <h6 class="text-subtitle1 q-my-none">
       return {
         tposs: [],
         currencyOptions: [
+          'satoshi',
```
Comment (motorina0) [inline templates/tpos/index.html:594]: Normally `sats` it used for `satoshis`. 
See env var here https://github.com/lnbits/lnbits/blob/dev/.env.example#L70

```env
LNBITS_DENOMINATION=sats
```

**Comment by motorina0** [comment]
> Thanks for the suggestions, @motorina0 . Hope I addressed them correctly. At least it still works when I test it. Also spotted one missed `${totalfsat}` to hide.

LGTM, can you please solve the conflict (`templates/tpos/index.html`)

**Comment by iWarpBTC** [comment]
Thanks for the heads up. Hadn't noticed. Done.

**Review by prusnak — APPROVED** [review]

**Pull request merged**